bitwarden_license/test/Scim.IntegrationTest/Controllers/v2/GroupsControllerTests.cs

```diff
@@ -0,0 +1,464 @@
+
+            Assert.Equal(StatusCodes.Status200OK, context.Response.StatusCode);
+
+            var databaseContext = _factory.GetDatabaseContext();
+            Assert.Equal(3, databaseContext.Groups.Count());
+
+            var firstGroup = databaseContext.Groups.FirstOrDefault(g => g.Id.ToString() == id);
+            Assert.Equal(newGroupName, firstGroup.Name);
+        }
+
+        [Fact]
+        public async Task Put_NotFound()
+        {
+            var newGroupName = "Test Group 1 New Name";
+            var organizationId = ScimApplicationFactory.TestOrganizationId1;
+            var id = Guid.NewGuid();
+            var model = new ScimGroupRequestModel
+            {
+                DisplayName = newGroupName,
+                ExternalId = "AA",
+                Members = new List<ScimGroupRequestModel.GroupMembersModel>(),
+                Schemas = new List<string>()
+            };
+
+            var context = await _factory.GroupsPutAsync(organizationId, id.ToString(), model);
+
+            Assert.Equal(StatusCodes.Status404NotFound, context.Response.StatusCode);
+
+            var responseModel = JsonSerializer.Deserialize<ScimErrorResponseModel>(context.Response.Body, new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase });
+            Assert.Equal(404, responseModel.Status);
+            Assert.Equal("Group not found.", responseModel.Detail);
+            Assert.Equal(new List<string> { ScimConstants.Scim2SchemaError }, responseModel.Schemas);
+
+            var databaseContext = _factory.GetDatabaseContext();
+            Assert.Equal(3, databaseContext.Groups.Count());
+            Assert.True(databaseContext.Groups.FirstOrDefault(g => g.Id == id) == null);
+        }
+
+        [Fact]
+        public async Task Patch_ReplaceDisplayName_Success()
+        {
+            var organizationId = ScimApplicationFactory.TestOrganizationId1;
+            var id = ScimApplicationFactory.TestGroupId1;
+            var model = new ScimPatchModel
+            {
+                Operations = new List<ScimPatchModel.OperationModel>()
+                {
+                    new ScimPatchModel.OperationModel
+                    {
+                        Op = "replace",
+                        Value = JsonDocument.Parse("{\"displayName\":\"Patch Display Name\"}").RootElement
+                    }
+                },
+                Schemas = new List<string>()
+            };
+
+            var context = await _factory.GroupsPatchAsync(organizationId, id, model);
+
+            Assert.Equal(StatusCodes.Status204NoContent, context.Response.StatusCode);
+
+            var databaseContext = _factory.GetDatabaseContext();
+            Assert.Equal(3, databaseContext.Groups.Count());
+
+            var group = databaseContext.Groups.FirstOrDefault(g => g.Id.ToString() == id);
+            Assert.Equal("Patch Display Name", group.Name);
```

Comment:
I think we should also assert the expected behaviour re: members. (I assume they are unaffected by this operation?)

Reply:
They're not but I've added a check to be sure.